src/features/filter/Filter.tsx

```diff
@@ -65,6 +68,24 @@ export const Filter = () => {
     setOpen(flag);
   };
 
+  const filterButton = useMemo(
+    () =>
+      isFiltered ? (
+        <Button
+          className="jdn__filter_filter-button"
+          type="link"
+          icon={
+            <Badge dot={true} color="blue" offset={[1, 4]}>
+              <Funnel size={14} color="#8C8C8C" />{" "}
+            </Badge>
+          }
+        />
+      ) : (
+        <Button className="jdn__filter_filter-button" type="link" icon={<Funnel size={14} color="#8C8C8C" />} />
+      ),
+    [isFiltered]
+  );
```

Comment:
I believe, that if ternary will be moved inside `icon={...}`, there will be less code duplication. Also `<Funnel size={14} color="#8C8C8C" />` can be added to a constant and reused